Develop stream 2025 05 06 (#542)
ROCm/rocThrust: CHANGELOG.md

Documentation for rocThrust available at
[https://rocm.docs.amd.com/projects/rocThrust/en/latest/](https://rocm.docs.amd.com/projects/rocThrust/en/latest/).

## rocThrust 4.0.0 for ROCm 7.0

### Changed

* Updated the required version of Google Benchmark from 1.8.0 to 1.9.0.

> * Drop `c++14` support for rocthrust.

Reviewer: I suggest putting this under "removed" rather than changed. Also, stating which version of C++ is currently supported is useful (I can also use it to update the documentation)

```suggestion
* C++14 support has been dropped.
```

Author: Yes, I have changed it to the same phrasing as rocprim.

> * Marked `thrust::device_malloc_allocator` as deprecated.

Reviewer: This would go under "upcoming changes" with a note saying that it'll be removed in upcoming versions.

```suggestion
* `thrust::device_malloc_allocator` is deprecated as of this version. It will be removed in an upcoming version.
```

Author: Change it.

> * Renamed `cpp14_required.h` to `cpp_version_check.h`

Reviewer: Is this something that the user needs to worry about? Because it feels like this would be a change that's transparent to the user. If it's not something the user needs to worry about, it doesn't have to be in the changelog.

```suggestion
* `cpp14_required.h` has been renamed to `cpp_version_check.h`
```

Author: I agree this probably should not be in here.

### Removed

> * Removed unused including header `device_malloc_allocator.h`.

Reviewer: I would argue that if it's unused it doesn't need to be here, but if you feel this is something a user needs to be aware of, then I'd rephrase it.

```suggestion
* `device_malloc_allocator.h` has been removed. This header file was unused and should not impact users.
```

Author: This could be used by users directly so better safe then sorry.
